src/Checkbox.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import PropTypes from 'prop-types'
+import { Icon, Flex, Box, Label, theme } from '..'
+
+class Checkbox extends React.Component {
+  handleClick(e) {
+    const { name, value, onClick } = this.props
+    const checked = e.target.checked
+    onClick(name, value, checked)
+  }
+
+  render() {
+    const {
+      name,
+      id,
+      disabled,
+      size,
+      fontSize,
+      children,
+      onClick,
+      checked,
+      defaultChecked,
+      theme: themeProp,
+      ...restProps
+    } = this.props
+    const currentTheme = themeProp || theme
+    return (
+      <StyledLabel
+        htmlFor={name}
+        data-test="design-system-checkbox"
+        disabled={disabled}
+      >
+        <input
+          type="checkbox"
+          name={name}
+          id={this.props.id || name}
+          disabled={disabled}
+          defaultChecked={checked || defaultChecked}
+          {...restProps}
+          onChange={e => this.handleClick(e)}
```

Comment:
I think we discussed this, but I think it's simpler to pass all props in to the `<input>` so that `onChange`, `onClick`, `onBlur`, etc. all work as expected – did you run into other issues with this since we last spoke?

Reply:
the main issue I have is that the display of the checkbox is separate from the actual element itself.  I have to change the display based on the native state of the checkbox, which I cannot reference without giving the component some kind of state itself